### PR TITLE
Ensure there are no fee transitions in executions

### PR DESCRIPTION
**Pull request by howardwu**
## Motivation

Ensure there are no fee transitions in executions

### CODE REVIEW

File: synthesizer/src/process/trace/mod.rs [hunk 1]
```diff
@@ -157,6 +157,14 @@ impl<N: Network> Trace<N> {
     ) -> Result<Execution<N>> {
         // Ensure this is not a fee.
         ensure!(!self.is_fee(), "The trace cannot prove execution for fee");
+        // Ensure there are no fee transitions.
+        ensure!(
+            self.transitions.iter().all(|transition| {
+                !(&transition.program_id().to_string() == "credits.aleo"
+                    && &transition.function_name().to_string() == "fee")
```
Comment (d0cd) [inline synthesizer/src/process/trace/mod.rs]:
```suggestion
                !transition.is_fee()
```

**Review by d0cd — COMMENTED** [review]
Minor nits

File: synthesizer/src/process/trace/mod.rs (hunk 1 above)
Comment (howardwu) [reply synthesizer/src/process/trace/mod.rs]: Fixed in [7e0551d](https://github.com/AleoHQ/snarkVM/pull/1648/commits/7e0551dc9bab1d6605488c37f7abb4920b209251)

**Review by d0cd — APPROVED** [review]
LGTM!

**Pull request merged**